roaring/roaring.go

```diff
@@ -1267,12 +1267,33 @@ func (c *container) bitmapToRun() {
 	// TODO
 }
 
+// arrayToRun converts from array format to RLE format
 func (c *container) arrayToRun() {
-	// TODO
+	c.runs = make([]interval32, 0)
+	start := c.array[0]
+	for i, v := range c.array {
+		if v - c.array[i-1] > 1 {
+			// if current-previous > 1, one run ends and another begins
+			c.runs = append(c.runs, interval32{start, c.array[i-1]})
+			start = v
+		}
+	}
+	// append final run
+	c.runs = append(c.runs, interval32{start, c.array[i]})
+	c.array = nil
+	c.mapped = false
```

Comment:
I'm not exactly sure about the implications of this, I copied it from the existing conversion functions